### PR TITLE
LUN-528: Delete deleted photos on saving

### PR DESCRIPTION
**Pull request by LeandraH**
This pull request belongs to the issue https://issues.tuerantuer.org/browse/LUN-528 on our bug tracker.

### CODE REVIEW

**Review by SteffiStuffel — CHANGES_REQUESTED** [review]
If a word is deleted the images should also be deleted.
Also if changing the article of a word, all existing and unmodified images are renamed. I think this does not really make sense.
Deleted images are correctly deleted :+1:

**Comment by sarahsporck** [comment]
@LeandraH somehow you have all the picker changes on this pr. Would you mind rebasing :)

**Review by SteffiStuffel — CHANGES_REQUESTED** [review]
Unfortunately this is not working, steps to reproduce:
1) Create a word with one image and save it
2) Create a word with three images and save it
3) Delete the first word (with one image)

Results:
Three images are deleted

Expected result:
Only the one image of the first word should be deleted

**Comment by LeandraH** [comment]
Ouch, good catch, thanks!

**Comment by SteffiStuffel** [comment]
> Ouch, good catch, thanks!

Thanks for fixing. Unfortunately this is still not working properly. Steps to reproduce:
1. Create a word with two images.
2. Save the word
3. Edit the word
4. delete one image
5. create a new image
6. save it

Expected results: One "old" image is there and one "new" image.

Actual results: Both old images are there the new images is not saved

**Comment by LeandraH** [comment]
Ooof, sorry, it should work now? I tested it quite a bit and couldn't find more problems 🤞

**Review by sarahsporck — APPROVED** [review]
Tested on Android emulator. Seems to be working fine. 👍🏼

File: src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx [hunk 1]
```diff
@@ -164,13 +176,17 @@ const UserVocabularyProcessScreen = ({ navigation, route }: UserVocabularyProces
...
+    setImages(images => images.filter(image => image !== uri))
+    setLocallyDeletedImages([...locallyDeletedImages, uri])
+  }
 
   if (showImageSelectionOverlay) {
     return <ImageSelectionOverlay setVisible={setShowImageSelectionOverlay} pushImage={pushImage} />
   }
 
   // TODO add Keyboard handling for input fields LUN-424
+  // TODO: find the new number
```
[the remark this answers is not in the document]
Comment (LeandraH) [reply src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx:189]: Sorry, WIP comment 🤦

File: src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx [hunk 2]
```diff
@@ -119,16 +120,29 @@ const UserVocabularyProcessScreen = ({ navigation, route }: UserVocabularyProces
...
         await incrementNextUserVocabularyId()
       }
 
       const imagePaths = await Promise.all(
         images.map(async (image, index) => {
-          const timestamp = Date.now()
-          const path = `file:///${DocumentDirectoryPath}/image-${id}-${index}-${timestamp}.jpg`
-          await moveFile(image, path)
+          let path: string
+          const imageHasBeenSavedPreviously = itemToEdit?.images.map(image => image.image).includes(image)
```
Comment (SteffiStuffel) [inline src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx:138]: naming this condition makes code much more readable :+1:

**Review by SteffiStuffel — APPROVED** [review]
Looks good, tested on Android :tada:

**Comment by SteffiStuffel** [comment]
Squashing commits would be nice

**Pull request merged**